aspnetcore/aspnetcore-2.1.md

```diff
@@ -0,0 +1,169 @@
+
+## [ApiController], ActionResult
+
+ASP.NET Core 2.1 add new programming conventions that make it easier to build clean and descriptive web APIs. `ActionResult<T>` is a new type added to allow an application to return either a response type or any other action result (similar to IActionResult), while still indicating the response type. The `[ApiController]` attribute has also been added as the way to opt in to Web API-specific conventions and behaviors.
+
+For more information, see [Build Web APIs with ASP.NET Core](xref:web-api/index)
+
+## IHttpClientFactory
+
+ASP.NET Core 2.1 includes a new `IHttpClientFactory` service that makes it easier to configure and consume instances of `HttpClient` in applications. `HttpClient` already has the concept of delegating handlers that could be linked together for outgoing HTTP requests. The factory:
+
+* Makes registering of instances of `HttpClient` per named client more intuitive.
+* Implements a Polly handler that allows Polly policies to be used for Retry, CircuitBreakers, etc.
+
+For more information, see [Initiate HTTP Requests](xref:fundamentals/http-requests)
+
+## Kestrel transport configuration
+
+With the release of ASP.NET Core 2.1, Kestrel's default transport is no longer based on Libuv but instead based on managed sockets. For more information, see [Kestrel web server implementation: Transport configuration](xref:fundamentals/servers/kestrel#transport-configuration).
+
+## Generic host builder
+
+In ASP.NET Core 2.1, the Generic Host Builder (`HostBuilder`) has been introduced. This builder can be used for applications that do not process HTTP requests (Messaging, background tasks, etc.).
+
+For more information, see [.NET Generic Host](xref:fundamentals/host/generic-host)
+
+## Updated SPA templates
+
+In ASP.NET Core 2.1, the Single Page Application templates for Angular, React, and React with Redux:
+
+* Are updated to use the standard project structures and build systems for each framework.
+
+The Angular template is based on Angular CLI, and the React templates are based on create-react-app.
+For more information, see [Use the Single Page Application templates with ASP.NET Core](xref:spa/index)
+
+## Razor Pages search for Razor assets
+
+In 2.1, Razor Pages search for Razor assets (such as layouts and partials) in to following directorys in the listed order:
+
+1. Current Pages folder.
+1. */Pages/Shared/*
+1. */Views/Shared/*
+
+## Razor Pages in an area
+
+Razor Pages now support [areas](xref:mvc/controllers/areas). To see an example of areas, create a new Razor Pages web app with individual user accounts. A Razor Pages web app with individual user accounts includes */Areas/Identity/Pages*.
+
+## Migrate from 2.0 to 2.1
+
+<!-- This needs to be updated to the specific doc -->
+See [Migrate from ASP.NET Core 2.0 to 2.1](xref:migration/index) and [SetCompatibilityVersion](xref:fundamentals/startup#setcompatibilityversion).
```

Comment:
I think we can remove the link to `SetCompatibilityVersion` here given that it's discussed already in the migration doc.